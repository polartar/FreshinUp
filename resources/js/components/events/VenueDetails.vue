<template>
  <v-card class="ff-venue_details">
    <v-card-title class="font-weight-bold">
      Venue Details
    </v-card-title>
    <v-divider />
    <v-card-text>
      <v-layout
        row
        wrap
        class="align-center"
      >
        <v-flex
          xs6
          pr-2
        >
          <v-layout
            row
            class="justify-content-between align-center mb-2"
          >
            <h4 class="text-uppercase grey--text font-weight-bold">
              Venue
            </h4>
            <clear-button
              v-if="location.venue_uuid"
              @clear="onVenueCleared"
            />
          </v-layout>
          <v-select
            :value="location.venue_uuid"
            :items="venues"
            item-text="name"
            item-value="uuid"
            single-line
            outline
            @input="onVenueChanged"
          />
        </v-flex>
        <v-flex
          xs6
          pl-2
        >
          <v-layout
            row
            class="justify-content-between align-center mb-2"
          >
            <h4 class="text-uppercase grey--text font-weight-bold">
              Location
            </h4>
            <clear-button
              v-if="location.uuid"
              @clear="onLocationCleared"
            />
          </v-layout>
          <v-select
            :value="location.uuid"
            :disabled="!location.venue_uuid"
            :items="selectedVenueLocations"
            item-text="name"
            item-value="uuid"
            single-line
            outline
            @input="onLocationChanged"
          />
        </v-flex>
        <v-flex
          xs12
          pb-3
          class="d-flex"
        >
          <h4 class="text-uppercase grey--text font-weight-bold">
            Address
          </h4>
          <p
            v-if="selectedVenue.address"
            class="grey--text text-xs-right"
          >
            {{ selectedVenue.address }}
          </p>
          <span
            v-else
            class="text-xs-center"
          >
            -
          </span>
        </v-flex>
        <v-flex xs12>
          <v-divider />
        </v-flex>
        <v-flex
          xs12
          py-3
          class="d-flex"
        >
          <h4 class="text-uppercase grey--text font-weight-bold">
            Capacity
          </h4>
          <p
            v-if="location.capacity"
            class="grey--text text-xs-right"
          >
            {{ location.capacity }} People
          </p>
          <span
            v-else
            class="text-xs-center"
          >
            -
          </span>
        </v-flex>
        <v-flex xs12>
          <v-divider />
        </v-flex>
        <v-flex
          xs12
          py-3
          class="d-flex"
        >
          <h4 class="text-uppercase grey--text font-weight-bold">
            Spots
          </h4>
          <p
            v-if="location.spots"
            class="grey--text text-xs-right"
          >
            {{ location.spots }}
          </p>
          <span
            v-else
            class="text-xs-center"
          >
            -
          </span>
        </v-flex>
        <v-flex xs12>
          <v-divider />
        </v-flex>
        <v-flex
          xs12
          py-3
        >
          <h4 class="pb-2 text-uppercase grey--text font-weight-bold">
            Location details
          </h4>
          <p v-if="hasLongText && !showMoreActivated">
            {{ minLocationDetail }}...
          </p>
          <p v-else>
            {{ location.details }}
          </p>
          <p
            v-if="hasLongText"
            class="text-xs-center"
          >
            <a
              href="#"
              class="grey--text font-weight-bold ff-venue_details__toggleLink"
              @click.prevent="toggleShowMore"
            >
              Show {{ showMoreActivated ? 'less' : 'more' }}
            </a>
          </p>
        </v-flex>
      </v-layout>
    </v-card-text>
    <v-card-actions>
      <v-layout
        align-center
        justify-space-around
      >
        <v-flex
          xs12
          sm8
          md6
          lg4
        >
          <v-btn
            disabled
            block
            depressed
            @click="getDirections"
          >
            Get Directions
          </v-btn>
        </v-flex>
      </v-layout>
    </v-card-actions>
  </v-card>
</template>
<script>
import ClearButton from '../ClearButton'
import omit from 'lodash/omit'
import get from 'lodash/get'
export const DEFAULT_LOCATION = {
  uuid: '',
  name: '',
  venue_uuid: '',
  capacity: '',
  spots: '',
  details: ''
}
export default {
  components: {
    ClearButton
  },
  props: {
    venues: { type: Array, default: () => [] },
    locationUuid: { type: String, default: '' },
    venueUuid: { type: String, default: '' }
  },
  data () {
    return {
      location: DEFAULT_LOCATION,
      showMoreActivated: false,
      locationDetailMaxChar: 300
    }
  },

  computed: {
    hasLongText () {
      return this.location.details.length > this.locationDetailMaxChar
    },
    minLocationDetail () {
      return this.location.details.slice(0, this.locationDetailMaxChar)
    },
    venuesByUuid () {
      return this.venues.reduce((map, venue) => {
        map[venue.uuid] = venue
        return map
      }, {})
    },
    selectedVenue () {
      return this.venuesByUuid[this.location.venue_uuid] || {}
    },
    selectedVenueLocations () {
      return get(this.selectedVenue, 'locations', [])
    },
    locationsByUuid () {
      return this.selectedVenueLocations.reduce((map, location) => {
        map[location.uuid] = location
        return map
      }, {})
    }
  },
  watch: {
    locationUuid (value) {
      this.onLocationChanged(value)
    },
    venueUuid (value) {
      this.onVenueChanged(value)
    },
    location (value) {
      this.$emit('input', value)
    }
  },
  mounted () {
    this.onVenueChanged(this.venueUuid)
    this.onLocationChanged(this.locationUuid)
  },

  methods: {
    toggleShowMore () {
      this.showMoreActivated = !this.showMoreActivated
    },
    getDirections () {
      this.$emit('get-directions', this.location)
    },
    onVenueCleared () {
      this.location.venue_uuid = ''
      this.onLocationCleared()
    },
    onVenueChanged (value) {
      this.onLocationCleared()
      this.location.venue_uuid = value
    },
    onLocationCleared () {
      this.location = Object.assign({}, this.location, omit(DEFAULT_LOCATION, ['venue_uuid']))
    },
    onLocationChanged (locationUuid) {
      const location = this.locationsByUuid[locationUuid] || DEFAULT_LOCATION
      this.location = Object.assign({}, this.location, omit(location, ['venue_uuid']))
    }
  }
}
</script>

<style>
  .ff-venue_details {
    --clear-button-size: 1rem;
  }

  .ff-venue_details__toggleLink {
    text-decoration: none;
  }
</style>
