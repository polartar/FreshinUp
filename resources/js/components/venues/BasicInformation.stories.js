import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

import BasicInformation from './BasicInformation'
import { FIXTURE_VENUE } from '../../../../tests/Javascript/__data__/venues'

export const Default = () => ({
  components: { BasicInformation },
  template: `
    <v-container>
      <basic-information/>
    </v-container>
  `
})

export const Loading = () => ({
  components: { BasicInformation },
  template: `
    <v-container>
      <basic-information loading/>
    </v-container>
  `
})

export const WithData = () => ({
  components: { BasicInformation },
  data () {
    return {
      venue: FIXTURE_VENUE
    }
  },
  template: `
    <v-container>
      <basic-information
        :value="venue"
        @input="onSave"
        @cancel="onCancel"
        @delete="onDelete"/>
    </v-container>
  `,
  methods: {
    onSave (payload) {
      action('onSave')(payload)
    },

    onCancel () {
      action('onCancel')()
    },

    onDelete (payload) {
      action('onDelete')(payload)
    }
  }
})

storiesOf('FoodFleet|components/venues/BasicInformation', module)
  .addParameters({
    backgrounds: [
      { name: 'default', value: '#f1f3f6', default: true }
    ]
  })
  .add('Default', Default)
  .add('Loading', Loading)
  .add('With data', WithData)
