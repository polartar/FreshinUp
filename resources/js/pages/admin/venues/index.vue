<template>
  <div>
    <v-flex
      d-flex
      align-center
      justify-space-between
      ma-3
    >
      <v-layout
        flex
        align-center
      >
        <h2 class="white--text mr-3">
          {{ pageTitle }}
        </h2>
        <v-btn
          slot="activator"
          color="white"
          @click="onCreateNew"
        >
          <span class="primary--text">Add New Venue</span>
        </v-btn>
      </v-layout>
    </v-flex>
    <v-divider />
    <template>
      <filter-sorter
        v-if="!isLoading"
        :statuses="statuses"
        @runFilter="filterVenues"
      />
      <venue-list
        :items="venues"
        :statuses="statuses"
        :is-loading="isLoading || isLoadingList"
        :rows-per-page="pagination.rowsPerPage"
        :page="pagination.page"
        :total-items="pagination.totalItems"
        :sort-by="sorting.sortBy"
        :descending="sorting.descending"
        @paginate="onPaginate"
        @manage-view="onManageView"
        @manage-delete="deleteSingle"
        @manage-multiple-delete="multipleDelete"
        @change-status="changeStatusSingle"
        @change-status-multiple="changeStatusMultiple"
      />
    </template>
    <v-dialog
      v-model="deleteDialog"
      max-width="500"
    >
      <simple-confirm
        :class="{ 'deleting': deletablesProcessing }"
        :title="deleteDialogTitle"
        ok-label="Yes"
        cancel-label="No"
        @ok="onSubmitDelete"
        @cancel="onCancelDelete"
      >
        <div class="py-5 px-2">
          <template v-if="deletablesProcessing">
            <div class="text-xs-center">
              <p class="subheading">
                Processing, please wait...
              </p>
              <v-progress-circular
                :rotate="-90"
                :size="200"
                :width="15"
                :value="deletablesProgress"
                color="primary"
              >
                {{ deletablesStatus }}
              </v-progress-circular>
            </div>
          </template>
          <template v-else>
            <p class="subheading">
              <span v-if="deletables.length < 2">Venue</span>
              <span v-else> Venues</span>
              : {{ deleteTemp | formatDeleteTitles }}
            </p>
          </template>
        </div>
      </simple-confirm>
    </v-dialog>
  </div>
</template>
<script>
import get from 'lodash/get'
import { mapActions, mapGetters, mapState } from 'vuex'
import { deletables } from 'fresh-bus/components/mixins/Deletables'
import VenueList from '~/components/venues/VenueList.vue'
import SimpleConfirm from 'fresh-bus/components/SimpleConfirm.vue'
import FilterSorter from '~/components/venues/FilterSorter'

const INCLUDE = [
  'status',
  'owner',
  'locations'
]

export default {
  layout: 'admin',
  components: {
    VenueList,
    SimpleConfirm,
    FilterSorter
  },
  filters: {
    formatDeleteTitles (value) {
      return value.map(item => item.name).join(', ')
    }
  },
  mixins: [deletables],
  data () {
    return {
      pageTitle: 'Venues List',
      deleteDialog: false,
      deleteTemp: [],
      deletablesProcessing: false,
      deletablesProgress: 0,
      deletablesStatus: '',
      lastFilterParams: {}
    }
  },
  computed: {
    isLoadingList () {
      return get(this.$store, 'state.venues.pending.items', true)
    },
    ...mapGetters(['currentUser']),
    ...mapGetters('venues', {
      venues: 'items',
      pagination: 'pagination',
      sorting: 'sorting',
      sortBy: 'sortBy'
    }),
    ...mapGetters('venueStatuses', { statuses: 'items' }),
    ...mapGetters('page', ['isLoading']),
    ...mapState('venues', ['sortables']),
    deleteDialogTitle () {
      return this.deleteTemp.length < 2 ? 'Are you sure you want to delete this venue?' : 'Are you sure you want to delete the following venues?'
    }
  },
  methods: {
    ...mapActions('page', {
      setPageLoading: 'setLoading'
    }),
    onCreateNew () {
      this.$router.push({ path: '/admin/venues/new' })
    },
    onManageView (venue) {
      this.$router.push({ path: `/admin/venues/${venue.uuid}/edit` })
    },
    deleteSingle (venue) {
      this.deleteTemp = [venue]
      this.deleteDialog = true
    },
    multipleDelete (venues) {
      this.deleteTemp = venues
      this.deleteDialog = true
    },
    changeStatusSingle (statusId, venue) {
      this.$store.dispatch('venues/patchItem', {
        data: { status_id: statusId },
        params: { id: venue.uuid }
      }).then(() => {
        this.filterVenues(this.lastFilterParams)
      })
    },
    changeStatusMultiple (statusId, venues) {
      venues.forEach((venue) => {
        this.changeStatusSingle(statusId, venue)
      })
    },
    async onSubmitDelete () {
      this.deletablesProcessing = true
      this.deletablesProgress = 0
      this.deletablesStatus = ''
      let dispatcheables = []

      this.deleteTemp.forEach((venue) => {
        dispatcheables.push(this.$store.dispatch('venues/deleteItem', {
          getItems: false,
          params: { id: venue.uuid }
        }))
      })

      let chunks = this.chunk(dispatcheables, this.deleteTempParrallelRequest)
      let doneCount = 0

      for (let i in chunks) {
        await Promise.all(chunks[i])
        doneCount += chunks[i].length
        this.deleteTempStatus = doneCount + ' / ' + this.deleteTemp.length + ' Done'
        this.deleteTempProgress = doneCount / this.deleteTemp.length * 100
        await this.sleep(this.deletablesSleepTime)
      }

      this.filterVenues(this.lastFilterParams)
      await this.sleep(500)
      this.deletablesProcessing = false
      this.deleteDialog = false
    },
    onCancelDelete () {
      this.deleteDialog = false
      this.deleteTemp = []
    },
    onPaginate (value) {
      this.$store.dispatch('venues/setPagination', value)
      this.$store.dispatch('venues/getItems', { params: { include: INCLUDE } })
    },
    filterVenues (params) {
      this.lastFilterParams = params
      this.$store.dispatch('venues/setFilters', {
        ...this.$route.query,
        ...this.lastFilterParams
      })
      this.$store.dispatch('venues/getItems', { params: { include: INCLUDE } })
    }
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.$store.dispatch('venues/setFilters', {
      ...vm.$route.query,
      ...this.lastFilterParams
    })
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('venueStatuses/getItems')
    ])
      .then(() => {
        if (next) next()
      })
      .catch(error => console.error(error))
      .then(() => {
        vm.setPageLoading(false)
      })
  }
}
</script>
