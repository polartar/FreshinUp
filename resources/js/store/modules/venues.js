import makeRestStore from 'fresh-bus/store/utils/makeRestStore'

export default ({ items, item }) => {
  return makeRestStore('foodfleet/venues', { items, item })
}
