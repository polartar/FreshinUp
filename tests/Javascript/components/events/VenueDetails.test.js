import { mount, shallowMount } from '@vue/test-utils'
import * as Stories from '~/components/events/VenueDetails.stories'
import Component, { DEFAULT_LOCATION } from '~/components/events/VenueDetails.vue'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import { FIXTURE_VENUES } from '../../__data__/venues'

describe('events/VenueDetails', () => {
  let localVue

  describe('Snapshots', () => {
    test('Default', async () => {
      const wrapper = mount(Stories.Default())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })

    test('With data', async () => {
      const wrapper = mount(Stories.WithData())
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Props & Computed', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
    })

    test('venues', () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.venues).toHaveLength(0)

      wrapper.setProps({
        venues: FIXTURE_VENUES
      })
      expect(wrapper.vm.venues).toEqual(FIXTURE_VENUES)
    })

    test('locationUuid', () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.locationUuid).toEqual('')

      const locationUuid = FIXTURE_VENUES[0].locations[0].uuid
      wrapper.setProps({
        locationUuid
      })
      expect(wrapper.vm.locationUuid).toEqual(locationUuid)
    })

    test('hasLongText', () => {
      const wrapper = shallowMount(Component)
      wrapper.setData({
        location: {
          details: FIXTURE_VENUES[0].locations[0].details
        },
        locationDetailMaxChar: 300
      })
      expect(wrapper.vm.hasLongText).toBe(false)
      wrapper.setData({
        locationDetailMaxChar: 50
      })
      expect(wrapper.vm.hasLongText).toBe(true)
    })

    test('minLocationDetail', () => {
      const wrapper = shallowMount(Component)
      wrapper.setData({
        location: {
          details: FIXTURE_VENUES[0].locations[0].details
        },
        locationDetailMaxChar: 50
      })
      expect(wrapper.vm.minLocationDetail).toHaveLength(50)
    })

    test('locations', async () => {
      const wrapper = shallowMount(Component)
      wrapper.setProps({
        venues: []
      })
      expect(wrapper.vm.locations).toHaveLength(0)

      wrapper.setProps({
        venues: FIXTURE_VENUES
      })
      wrapper.setData({
        venueUuid: FIXTURE_VENUES[1].uuid
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.locations).toHaveLength(FIXTURE_VENUES[1].locations.length)
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
    })

    test('onLocationCleared()', () => {
      const wrapper = shallowMount(Component)
      wrapper.setData({
        location: {
          uuid: 'abc123'
        }
      })
      wrapper.vm.onLocationCleared()
      expect(wrapper.vm.location).toMatchObject(DEFAULT_LOCATION)
    })

    test('onVenueCleared()', () => {
      const wrapper = shallowMount(Component)
      wrapper.setData({
        location: {
          uuid: 'abc123',
          venue_uuid: 'venue_abc123'
        }
      })
      wrapper.vm.onVenueCleared()
      expect(wrapper.vm.location.venue_uuid).toBeNull()
      expect(wrapper.vm.location).toMatchObject(DEFAULT_LOCATION)
    })

    test('onVenueChanged(value)', () => {
      const wrapper = shallowMount(Component)
      wrapper.setData({
        location: {
          uuid: 'abc123',
          venue_uuid: 'venue_abc111'
        }
      })
      wrapper.vm.onVenueChanged('venue_abc222')
      expect(wrapper.vm.location).toMatchObject({ ...DEFAULT_LOCATION, venue_uuid: 'venue_abc222' })
    })

    test('onLocationChanged(value)', async () => {
      const wrapper = shallowMount(Component)
      wrapper.setData({
        location: FIXTURE_VENUES[0].locations[0],
        venues: FIXTURE_VENUES
      })
      wrapper.vm.onLocationChanged(FIXTURE_VENUES[0].locations[1].uuid)
      await wrapper.vm.$nextTick()
      await wrapper.vm.$nextTick()
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.location).toMatchObject(FIXTURE_VENUES[0].locations[1])
    })

    test('toggleShowMore()', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue
      })

      wrapper.setData({
        showMoreActivated: false
      })

      wrapper.vm.toggleShowMore()
      expect(wrapper.vm.showMoreActivated).toBe(true)
      wrapper.vm.toggleShowMore()
      expect(wrapper.vm.showMoreActivated).toBe(false)
    })
    describe('getDirections()', () => {
      test('when empty venue data', async () => {
        const wrapper = shallowMount(Component)
        wrapper.vm.getDirections()
        await wrapper.vm.$nextTick()
        const emitted = wrapper.emitted()
        expect(emitted['get-directions']).toHaveLength(1)
        expect(emitted['get-directions'][0][0]).toEqual(DEFAULT_LOCATION)
      })
      test('when venue data is set', async () => {
        const wrapper = shallowMount(Component)
        const location = FIXTURE_VENUES[0].locations[0]
        wrapper.setData({
          location
        })
        await wrapper.vm.$nextTick()
        wrapper.vm.getDirections()
        await wrapper.vm.$nextTick()
        const emitted = wrapper.emitted()
        expect(emitted['get-directions']).toHaveLength(1)
        expect(emitted['get-directions'][0][0]).toEqual(location)
      })
    })
  })
})
