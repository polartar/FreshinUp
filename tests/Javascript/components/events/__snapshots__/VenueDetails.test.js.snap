// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`events/VenueDetails Snapshots Default 1`] = `
<div
  class="container"
>
  <div
    class="ff-venue_details v-card v-sheet theme--light"
  >
    <div
      class="v-card__title font-weight-bold"
    >
      
    Venue Details
  
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      class="v-card__text"
    >
      <div
        class="layout align-center row wrap"
      >
        <div
          class="flex xs6 pr-2"
        >
          <div
            class="layout justify-content-between align-center mb-2 row"
          >
            <h4
              class="text-uppercase grey--text font-weight-bold"
            >
              
            Venue
          
            </h4>
             
            <!---->
          </div>
           
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <div
                    class="v-select__selections"
                  >
                    <input
                      aria-readonly="false"
                      autocomplete="on"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <div
                            class="v-list__tile theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                No data available
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex xs6 pl-2"
        >
          <div
            class="layout justify-content-between align-center mb-2 row"
          >
            <h4
              class="text-uppercase grey--text font-weight-bold"
            >
              
            Location
          
            </h4>
             
            <!---->
          </div>
           
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select v-input--is-disabled theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <div
                    class="v-select__selections"
                  >
                    <input
                      aria-readonly="false"
                      autocomplete="on"
                      disabled="disabled"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon v-icon--disabled material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <div
                            class="v-list__tile theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                No data available
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex d-flex xs12 pb-3"
        >
          <h4
            class="text-uppercase grey--text font-weight-bold"
          >
            
          Address
        
          </h4>
           
          <span
            class="text-xs-center"
          >
            
          -
        
          </span>
        </div>
         
        <div
          class="flex xs12"
        >
          <hr
            class="v-divider theme--light"
          />
        </div>
         
        <div
          class="flex d-flex xs12 py-3"
        >
          <h4
            class="text-uppercase grey--text font-weight-bold"
          >
            
          Capacity
        
          </h4>
           
          <span
            class="text-xs-center"
          >
            
          -
        
          </span>
        </div>
         
        <div
          class="flex xs12"
        >
          <hr
            class="v-divider theme--light"
          />
        </div>
         
        <div
          class="flex d-flex xs12 py-3"
        >
          <h4
            class="text-uppercase grey--text font-weight-bold"
          >
            
          Spots
        
          </h4>
           
          <span
            class="text-xs-center"
          >
            
          -
        
          </span>
        </div>
         
        <div
          class="flex xs12"
        >
          <hr
            class="v-divider theme--light"
          />
        </div>
         
        <div
          class="flex xs12 py-3"
        >
          <h4
            class="pb-2 text-uppercase grey--text font-weight-bold"
          >
            
          Location details
        
          </h4>
           
          <p>
            
          
        
          </p>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="v-card__actions"
    >
      <div
        class="layout align-center justify-space-around"
      >
        <div
          class="flex xs12 sm8 md6 lg4"
        >
          <button
            class="v-btn v-btn--block v-btn--disabled v-btn--depressed theme--light"
            disabled="disabled"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Get Directions
        
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`events/VenueDetails Snapshots With data 1`] = `
<div
  class="container"
>
  <div
    class="ff-venue_details v-card v-sheet theme--light"
  >
    <div
      class="v-card__title font-weight-bold"
    >
      
    Venue Details
  
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      class="v-card__text"
    >
      <div
        class="layout align-center row wrap"
      >
        <div
          class="flex xs6 pr-2"
        >
          <div
            class="layout justify-content-between align-center mb-2 row"
          >
            <h4
              class="text-uppercase grey--text font-weight-bold"
            >
              
            Venue
          
            </h4>
             
            <!---->
          </div>
           
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select v-input--is-label-active v-input--is-dirty theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <div
                    class="v-select__selections"
                  >
                    <div
                      class="v-select__selection v-select__selection--comma"
                    >
                      LA Stadium
                    </div>
                    <input
                      aria-readonly="false"
                      autocomplete="on"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          class="primary--text"
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                LA Stadium
                              </div>
                            </div>
                          </a>
                        </div>
                        <div
                          role="listitem"
                        >
                          <a
                            class="v-list__tile v-list__tile--link theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                ATL Stadium
                              </div>
                            </div>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex xs6 pl-2"
        >
          <div
            class="layout justify-content-between align-center mb-2 row"
          >
            <h4
              class="text-uppercase grey--text font-weight-bold"
            >
              
            Location
          
            </h4>
             
            <!---->
          </div>
           
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select v-input--is-disabled theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <div
                    class="v-select__selections"
                  >
                    <input
                      aria-readonly="false"
                      autocomplete="on"
                      disabled="disabled"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon v-icon--disabled material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <div
                            class="v-list__tile theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                No data available
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex d-flex xs12 pb-3"
        >
          <h4
            class="text-uppercase grey--text font-weight-bold"
          >
            
          Address
        
          </h4>
           
          <span
            class="text-xs-center"
          >
            
          -
        
          </span>
        </div>
         
        <div
          class="flex xs12"
        >
          <hr
            class="v-divider theme--light"
          />
        </div>
         
        <div
          class="flex d-flex xs12 py-3"
        >
          <h4
            class="text-uppercase grey--text font-weight-bold"
          >
            
          Capacity
        
          </h4>
           
          <span
            class="text-xs-center"
          >
            
          -
        
          </span>
        </div>
         
        <div
          class="flex xs12"
        >
          <hr
            class="v-divider theme--light"
          />
        </div>
         
        <div
          class="flex d-flex xs12 py-3"
        >
          <h4
            class="text-uppercase grey--text font-weight-bold"
          >
            
          Spots
        
          </h4>
           
          <span
            class="text-xs-center"
          >
            
          -
        
          </span>
        </div>
         
        <div
          class="flex xs12"
        >
          <hr
            class="v-divider theme--light"
          />
        </div>
         
        <div
          class="flex xs12 py-3"
        >
          <h4
            class="pb-2 text-uppercase grey--text font-weight-bold"
          >
            
          Location details
        
          </h4>
           
          <p>
            
          
        
          </p>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="v-card__actions"
    >
      <div
        class="layout align-center justify-space-around"
      >
        <div
          class="flex xs12 sm8 md6 lg4"
        >
          <button
            class="v-btn v-btn--block v-btn--disabled v-btn--depressed theme--light"
            disabled="disabled"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
          Get Directions
        
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
